### PR TITLE
Test for pet query and addition of non-nullable attributes for pet and application type. Remade json schema file.

### PR DESCRIPTION
**Pull request by kbhoffmann**
What changed?
Made all fields in Pet type and Application non-nullable except for owner_story in PetType.
Regeneraged schema.graphql and schema.json files to reflect non-nullable updates.
Remove .strip from not_furgotten schema test for .to_definition.
Start of spec file for query for a pet by id.  More tests in progress.

Why is this change necessary?
Reflect discussion with FE team for fields user can leave blank in form/ null values allowed in database.

What changed technically that may impact others?
Update in schema.graphql and schema.json files may impact FE queries, but we have not given them that contract yet.

How do we test it?
**bundle exec rspec**     to test the schema files
**bundle exec rspec spec/graphql/queries/pets/get_pet_by_id.rb  **   to test the query for get_pet_by_id

Something may be incorrect in our spec file tree that will have to be changed so bundle exec rspec runs all tests.

### CODE REVIEW

**Review by omegaeye-he — COMMENTED** [review]
This PR is huge. It can be split up into multiple PRs. You're changing up `application_type`, `pet_type`, `query_type`, `schema`, ``not_forgotten_schema_spec`, `get_all_pets_spec`, and `get_pet_by_id` all in one go. The person who reviews this code has to have all the contexts for every file in order to follow.

File: schema.graphql [hunk 1]
```diff
@@ -1,10 +1,10 @@
 type Application {
   createdAt: ISO8601DateTime!
-  description: String
-  email: String
+  description: String!
```
Comment (karanm645) [inline schema.graphql:3]: Curious what the bang does after the type

File: schema.json [hunk 2]
```diff
@@ -510,29 +547,36 @@
...
+                {
+                  "name": "id",
+                  "description": null,
+                  "type": {
                     "kind": "NON_NULL",
                     "name": null,
                     "ofType": {
-                      "kind": "OBJECT",
-                      "name": "Pet",
+                      "kind": "SCALAR",
```
Comment (karanm645) [inline schema.json:571]: What kind of value does SCALAR represent?

File: spec/graphql/not_furgotten_schema_spec.rb [hunk 3]
```diff
@@ -3,9 +3,9 @@
 RSpec.describe NotFurgottenSchema do
   it 'matches the dumped schema (rails graphql:schema:dump)' do
     aggregate_failures do
-      #without rstrip n line 7 it fails because of whitespace
-      expect(described_class.to_definition.rstrip).to eq(File.read(Rails.root.join('schema.graphql')).rstrip)
```
Comment (karanm645) [inline spec/graphql/not_furgotten_schema_spec.rb:7]: Nice 👽

**Review by karanm645 — APPROVED** [review]
Looks good! Just have a few minor questions but all in all great work!!!!

**Review by croixk — APPROVED** [review]
Looks good! Good work!

**Pull request merged**